cognite/client/data_classes/geospatial.py

```diff
@@ -0,0 +1,182 @@
+from typing import Any, Dict
+
+from cognite.client import utils
+from cognite.client.data_classes._base import CogniteResource, CogniteResourceList
+
+
+class FeatureType(CogniteResource):
+    """A representation of a feature type in the geospatial api."""
+
+    def __init__(
+        self,
+        external_id: str = None,
+        created_time: int = None,
+        last_updated_time: int = None,
+        properties: Dict[str, Any] = None,
+        search_spec: Dict[str, Any] = None,
+        cognite_client=None,
+    ):
+        self.external_id = external_id
+        self.created_time = created_time
+        self.last_updated_time = last_updated_time
+        self.properties = properties
+        self.search_spec = search_spec
+        self._cognite_client = cognite_client
+
+    @classmethod
+    def _load(cls, resource: Dict, cognite_client=None):
+        instance = cls(cognite_client=cognite_client)
+        for key, value in resource.items():
+            snake_case_key = utils._auxiliary.to_snake_case(key)
+            setattr(instance, snake_case_key, value)
+        return instance
+
+
+class FeatureTypeList(CogniteResourceList):
+    _RESOURCE = FeatureType
+    _ASSERT_CLASSES = False
+
+
+class PropertyAndSearchSpec:
+    """A representation of a feature type property and search spec."""
+
+    def __init__(self, properties: Dict[str, Any] = None, search_spec: Dict[str, Any] = None):
+        self.properties = properties
+        self.search_spec = search_spec
+
+
+class FeatureTypeUpdate:
+    """A representation of a feature type update in the geospatial api."""
+
+    def __init__(self, external_id: str = None, add: PropertyAndSearchSpec = None, cognite_client=None):
+        self.external_id = external_id
+        self.add = add
+        self._cognite_client = cognite_client
+
+
+class FeatureTypeUpdateList:
+    _RESOURCE = FeatureTypeUpdate
+    _ASSERT_CLASSES = False
+
+
+class Feature(CogniteResource):
+    """A representation of a feature in the geospatial api."""
+
+    def __init__(self, external_id: str = None, cognite_client=None, **properties):
```

Comment:
What is `properties` here? Arbitrary key-value pairs?

Reply:
not arbitrary, they have to conform to the feature type definition

Reply:
I see.